app/src/main/java/com/example/dine_and_donate/HomeFragments/MapFragment.java

```diff
@@ -698,6 +529,145 @@ public void onComplete(@Nullable DatabaseError databaseError, @NonNull DatabaseR
         }
     }
 
+    // LOAD MAP AND GET LOCATION //
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    @Override
+    public void onMapReady(GoogleMap googleMap) {
+        map = googleMap;
+        map.setOnCameraIdleListener(new GoogleMap.OnCameraIdleListener() {
+            @Override
+            public void onCameraIdle() {
+                Double newLongitude = map.getCameraPosition().target.longitude;
+                Double newLatitude = map.getCameraPosition().target.latitude;
+                if (cameraLatitude == null || cameraLongitude == null) {
+                    cameraLatitude = newLatitude;
+                    cameraLongitude = newLongitude;
+
+                    if (mCurrentUser.isOrg) {
+                        generateMarkersRestaurants(Double.toString(cameraLongitude), Double.toString(cameraLatitude));
+                    } else {
+                        generateMarkersEvents();
+                    }
+                }
+
+                if (map.getCameraPosition().zoom > 10) {
+                    cameraLongitude = newLongitude;
+                    cameraLatitude = newLatitude;
+
+                    if (mCurrentUser.isOrg) {
+                        generateMarkersRestaurants(Double.toString(cameraLongitude), Double.toString(cameraLatitude));
+                    } else {
+                        generateMarkersEvents();
+                    }
+                }
+            }
+        });
+
+        if (map != null) {
+            // Map is ready
+            MapDemoFragmentPermissionsDispatcher.getMyLocationWithPermissionCheck(this);
+            MapDemoFragmentPermissionsDispatcher.startLocationUpdatesWithPermissionCheck(this);
+
+            //map.setOnMapLongClickListener(this);
+            map.setInfoWindowAdapter(new CustomWindowAdapter(getLayoutInflater()));
+
+            map.setMapType(GoogleMap.MAP_TYPE_TERRAIN);
+        } else {
+            Toast.makeText(mContext, "Error - Map was null!!", Toast.LENGTH_SHORT).show();
```

Comment:
I don't think you should display this to the user. Put it in a Log.e statement instead. If you want to show something to the user, just say "Sorry, something went wrong" in the toast.